Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -1308,8 +1304,8 @@ void TextureMapperLayer::setSize(const FloatSize& size)
     if (canInferDamage() && m_state.size != size) {
         // When layer size changes, we damage whole layer (before and after) for now.
         // FIXME: Damage only affected area.
-        m_inferredLayerDamage.add(FloatRect(FloatPoint::zero(), m_state.size));
-        m_inferredLayerDamage.add(FloatRect(FloatPoint::zero(), size));
+        m_damageInLayerCoordinateSpace.add(FloatRect(FloatPoint::zero(), m_state.size));
```

Comment:
re-reading the comment above, something we might do is extracting this logic into a separate method. Something like

```
const auto& layerResizedDamage = getLayerResizedDamage(/* maybe passing in old and new */);
for (const auto& damage : layerResizedDamage)
    m_damageInLayerCoordinateSpace.add(damage);
```

then algorithmically this method (`setSize`) becomes correct, and it's only the new method that needs a `notImplemented()` + the default, more expensive, implementation you suggested above

Reply:
I think this will reduce both readability and performance, so I'm not sure